internal/tokens/tokens.go

```diff
@@ -0,0 +1,158 @@
+package tokens
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/skratchdot/open-golang/open"
+	"github.com/superfly/flyctl/internal/logger"
+	"github.com/superfly/flyctl/iostreams"
+	"github.com/superfly/macaroon"
+	"github.com/superfly/macaroon/flyio"
+	"github.com/superfly/macaroon/tp"
+	"golang.org/x/exp/slices"
+)
+
+type Tokens struct {
+	macaroonTokens []string
+	userTokens     []string
+}
+
+func Parse(token string) *Tokens {
+	token = stripAuthorizationScheme(token)
+	ret := &Tokens{}
+
+	for _, tok := range strings.Split(token, ",") {
+		tok = strings.TrimSpace(tok)
+		switch pfx, _, _ := strings.Cut(tok, "_"); pfx {
+		case "fm1r", "fm1a", "fm2":
+			ret.macaroonTokens = append(ret.macaroonTokens, tok)
+		default:
+			ret.userTokens = append(ret.userTokens, tok)
+		}
+	}
+
+	return ret
+}
+
+func (t *Tokens) Flaps() string {
+	return t.normalized(false)
+}
+
+func (t *Tokens) Docker() string {
+	return t.normalized(false)
+}
+
+func (t *Tokens) NATS() string {
+	return t.normalized(false)
+}
+
+func (t *Tokens) API() string {
```

Comment:
- Is API means `web`? Flaps is also having API...
- Does it make sense to have `WithUserToken` instead? It is a bit odd to see the tokens package know all use cases inside flyctl.

Reply:
> Does API here means web? Flaps is also having API...

Good point. I can come up with a better name.

> Does it make sense to have WithUserToken instead? It is a bit odd to see the tokens package knows all use cases inside flyctl.

That's what I did originally. I felt like it would be easier for future developers to use the flaps token for flaps, the docker token for docker, etc... rather than asking them to think about whether they want the user token or not. I'm fine either way though if you think `WithUserToken` is a nicer API.